packages/flutter/lib/src/painting/colors.dart

```diff
@@ -71,10 +75,52 @@ class HSVColor {
       b = x;
     }
     return new Color.fromARGB(
-      ( a      * 0xFF).round(),
+      (alpha   * 0xFF).round(),
       ((r + m) * 0xFF).round(),
       ((g + m) * 0xFF).round(),
       ((b + m) * 0xFF).round()
     );
   }
+
+  /// Linearly interpolate between two HSVColors.
+  ///
+  /// If either color is null, this function linearly interpolates from a
+  /// transparent instance of the other color.
+  static HSVColor lerp(HSVColor a, HSVColor b, double t) {
+    if (a == null && b == null)
+      return null;
+    if (a == null)
+      return _scaleAlpha(b, t);
+    if (b == null)
+      return _scaleAlpha(a, 1.0 - t);
+    return new HSVColor.fromAHSV(
+      lerpDouble(a.alpha, b.alpha, t),
+      lerpDouble(a.hue, b.hue, t),
+      lerpDouble(a.saturation, b.saturation, t),
+      lerpDouble(a.value, b.value, t)
+    );
+  }
+
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other is! HSVColor)
+      return false;
+    final HSVColor typedOther = other;
+    return typedOther.alpha == alpha
```

Comment:
We should be consistent about how we do && and || cascades. This and the || cascade in the patch earlier today or yesterday are the first times we've done it that way. They do look better in "return" cascades than what we normally do, though. Maybe we should say the rule is "in a return expression, if you have a sequence of subexpressions are delimited by && or ||, then put the logical operator under the return; in all other cases, when you line break at a logical operator, put it at the end of the line".

If you agree, file a bug to make the code consistent and to update the style guide.